src/mumble/ConnectDialog.h

```diff
@@ -116,6 +116,7 @@ class ServerItem : public QTreeWidgetItem, public PingStats {
 		enum ItemType { FavoriteType, LANType, PublicType };
 
 		static QMap<QString, QIcon> qmIcons;
+		static QString filter;
```

Comment:
I have a feeling that the approach with the static filter string is not the best.
Just the code in `ConnectDialog::on_qleServerFilter_textChanged` wasn't enough because it would filter all the servers that were already in the list, but not the ones that were populating the list _after_ the text was entered.

I don't have too much experience so feedback would be appreciated.

Reply:
You could catch where a new ServerItem gets created and then apply the filter before adding it to the list. For that you might have to add a public getter for the filter's text.
Alternatively you could trigger a change even for the filter box every time a new item was added and by that basically pretending as if that had changed its text :thinking: